frontend/src/api/task.ts

```diff
@@ -0,0 +1,68 @@
+import { post } from "src/api/requests";
+
+/**
+ * Initializes the shape for the data being retrieved from the backend.
+ */
+export interface Task {
+  _id: string;
+  title: string;
+  description?: string;
+  isChecked: boolean;
+  dateCreated: Date;
+}
+
+interface TaskJSON {
+  _id: string;
+  title: string;
+  description?: string;
+  isChecked: boolean;
+  dateCreated: string;
+}
+
+/**
+ * Converts taks from "any"-typed JSON that only contains primitive types to our
+ * custom Task interface. This is used to change the date field from a string
+ * to a Date object.
+ *
+ * @param task the JSON representation of the task
+ * @returns the typed Task
+ */
+function parseTask(task: TaskJSON): Task {
+  return {
+    _id: task._id,
+    title: task.title,
+    description: task.description,
+    isChecked: task.isChecked,
+
+    // convert the string version of the date to a Date object
+    dateCreated: new Date(task.dateCreated),
```

Comment:
This seems obvious enough lol

```suggestion
    dateCreated: new Date(task.dateCreated),
```